libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/ptx/ptx_helper_functions.h

```diff
@@ -43,6 +59,46 @@ inline _LIBCUDACXX_DEVICE _CUDA_VSTD::uint64_t __as_ptr_gmem(const void* __ptr)
   return static_cast<_CUDA_VSTD::uint64_t>(__cvta_generic_to_global(__ptr));
 }
 
+
+/*************************************************************
+ *
+ * Conversion from state space "pointer" -> generic pointer
+ *
+ **************************************************************/
+template <typename _Tp>
+inline _LIBCUDACXX_DEVICE _Tp* __from_ptr_smem(_CUDA_VSTD::size_t __ptr)
+{
+  // Consider adding debug asserts here.
+  return reinterpret_cast<_Tp*>(__cvta_shared_to_generic(__ptr));
+}
+
+template <typename _Tp>
+inline _LIBCUDACXX_DEVICE _Tp* __from_ptr_dsmem(_CUDA_VSTD::size_t __ptr)
+{
+  // Consider adding debug asserts here.
+  return reinterpret_cast<_Tp*>(__cvta_shared_to_generic(__ptr));
```

Comment:
Applies throughout. If there are no differences, then implement once and delegate